mlflow/tracking/client.py

```diff
@@ -610,10 +610,10 @@ def start_trace(
             if isinstance(mlflow_span, NoOpSpan):
                 return mlflow_span
 
-            if inputs:
+            if inputs is not None:
                 mlflow_span.set_inputs(inputs)
-            if attributes:
-                mlflow_span.set_attributes(attributes)
+            mlflow_span.set_attributes(attributes or {})
```

Comment:
note: it's fine to do implicit boolean casting for attributes, because it is always dictionary or none (we have type check in `set_attribtues`).